services/codecov/codecov.service.js

```diff
@@ -1,108 +1,139 @@
 'use strict'
 
-const queryString = require('query-string')
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const {
-  coveragePercentage: coveragePercentageColor,
-} = require('../color-formatters')
+const Joi = require('joi')
+const { coveragePercentage } = require('../color-formatters')
+const { BaseJsonService } = require('..')
 
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Codecov extends LegacyService {
+// https://docs.codecov.io/reference#totals
+// A new repository that's been added but never had any coverage reports
+// uploaded will not have a `commit` object in the response and some times
+// the `totals` object can also be missing for the latest commit.
+// Keeping these as optional in the schema and handling it in the transform
+// function to maintain consistent badge behavior with the legacy service implementation.
```

Comment:
Working on updating this. Initially this was another one of those "because that's the way it was done before in the legacy service" things 😄 

However, I think the underlying scenarios that the legacy service implementation was trying to support was valid (non error) user cases when the coverage was not yet known. 

I can try to dig into it a bit more